cmd/juju/commands/run.go

```diff
@@ -101,10 +106,16 @@ func (c *runCommand) SetFlags(f *gnuflag.FlagSet) {
 		"default": cmd.FormatYaml,
 	})
 	f.BoolVar(&c.all, "all", false, "Run the commands on all the machines")
-	f.DurationVar(&c.timeout, "timeout", 5*time.Minute, "How long to wait before the remote command is considered to have failed")
+	for _, flag := range[]string{"timeout", "t"} {
```

Comment:
I made a mistake here in my original review comment.
You don't need the for loop. You just need to start with the short arg "t" and then use empty values for the subsequent longer arg,

f.DurationVar(&c.timeout, "t", 5*time.Minute, "How long to wait before the remote command is considered to have failed")
f.DurationVar(&c.timeout, "timeout", 0, "")

Same with unit, application etc